File: py/grading_students.py
def grading_students(grades):
    finalgrades = []

    for grade in grades:
        if grade < 38 or grade % 5 == 0 or grade % 5 < 3:
            finalgrades.append(grade)
        elif grade % 5 > 2:
            finalgrades.append(grade + (5 - (grade % 5)))

    return finalgrades

fg = grading_students([73, 67, 38, 33])
print(fg)

File: cs/gradingstudents.cs
static List<int> gradingStudents(List<int> grades)
{
    List<int> finalgrades = new List<int>();
    foreach (int grade in grades)
        if(grade < 38 || grade % 5 == 0 || grade % 5 < 3)
            finalgrades.Add(grade);
        else if(grade % 5 > 2)
            finalgrades.Add(grade + (5- (grade % 5)));
    
    return finalgrades;
}

List<int> g = new List<int>(new int[] { 73, 67, 38, 33 } );
var fg = gradingStudents(g);

Console.Write(string.Join(" ", fg));